### ISSUE TITLE
The edge_index in HeteoGraph with nx.Graph is not correct when running in message_passing 

### ISSUE THREAD

**Issue by dengfy** [snap-stanford/deepsnap#7]
the order with source node and target node will exchange.

`
if self.isdirect():
    if isinstance(edge_index, dict):
        for key in edge_index:
            edge_index[key] = torch.cat(
                [edge_index[key], torch.flip(edge_index[key], [1])], dim=0)

if isinstance(edge_index, dict):
    for key in edge_index:
        permute_tensor = edge_index[key].permute(1, 0)
        source_node_index = \
            _convert_to_tensor_index(permute_tensor[0])
        target_node_index = \
            _convert_to_tensor_index(permute_tensor[1])
        edge_index[key] = \
            torch.stack([source_node_index, target_node_index])`

**Comment by RexYing** [comment]
Hi could you clarify a bit what do you mean by exchange?
The main purpose of the first part of this code is to handle the undirected case where we flip the edge direct so message passing can go in both directions.

The purpose of the second part you pasted is to convert the tensor from E x 2 to 2 x E, and the first row is the first column before, which is the source node. These 2 parts serve distinct purposes.
